iothub/device/samples/how to guides/DeviceReconnectionSample/CustomRetryPolicy.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    /// <summary>
+    /// An exponential backoff with jitter based retry policy that retries on additional exceptions
+    /// not covered by the SDK, because in this application we wish to run as long as possible.
+    /// </summary>
+    internal class CustomRetryPolicy : IRetryPolicy
```

Comment:
Rename to make it clearer that we're customizing, rather than name it what it is. We'll rely on comments to make that clearer.